app/uk/gov/hmrc/soletraderidentificationfrontend/services/SubmissionService.scala

```diff
@@ -25,61 +25,163 @@ import javax.inject.{Inject, Singleton}
-          Future.successful(JourneyCompleted(journeyConfig.continueUrl))
-        case Right(_) if individualDetails.optNino.isEmpty && isEnabled(EnableNoNinoJourney) => for {
-          _ <- createTrnService.createTrn(journeyId)
-          _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
-          _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
-        } yield {
-          JourneyCompleted(journeyConfig.continueUrl)
-        }
-        case Right(_) => for {
-          _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
-          _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
-        } yield {
-          JourneyCompleted(journeyConfig.continueUrl)
-        }
-        case Left(failureReason) =>
+    } yield
+      response
+
+  private def noDataServerException(journeyId: String): Nothing =
+    throw new InternalServerException(s"Details could not be retrieved from the database for $journeyId")
+
+  private def calculateBusinessScenario(journeyConfig: JourneyConfig): BusinessScenario =
+    if (!journeyConfig.pageConfig.enableSautrCheck) IndividualJourney
+    else {
+      if (journeyConfig.businessVerificationCheck) SoleTraderJourneyWithSaUtrCheck
+      else SoleTraderJourneySkippingSaUtrCheck
+    }
+
+  private def calculateSubmissionResponse(journeyId: String,
+                                          businessScenario: BusinessScenario,
+                                          matchingResult: Either[SoleTraderDetailsMatching.SoleTraderDetailsMatchFailure, Boolean],
+                                          journeyConfig: JourneyConfig,
+                                          individualDetails: IndividualDetails)(implicit hc: HeaderCarrier,
+                                                                                ec: ExecutionContext): Future[SubmissionResponse] = businessScenario match {
+    case SoleTraderJourneySkippingSaUtrCheck =>
+      handleSoleTraderJourneySkippingBVCheck(journeyId, matchingResult, journeyConfig.continueUrl, individualDetails.optSautr, individualDetails.optNino)
+    case SoleTraderJourneyWithSaUtrCheck =>
+      handleSoleTraderJourneyWithBVCheck(journeyId, matchingResult, journeyConfig, individualDetails)
+    case IndividualJourney =>
+      handleIndividualJourney(journeyId, matchingResult, journeyConfig.continueUrl)
+  }
+
+  private def handleSoleTraderJourneySkippingBVCheck(journeyId: String,
+                                                     matchingResult: Either[SoleTraderDetailsMatching.SoleTraderDetailsMatchFailure, Boolean],
+                                                     continueUrl: String,
+                                                     maybeSaUtr: Option[String],
+                                                     maybeNino: Option[String]
```

Comment:
Could we please change these to optSautr and optNino - just to keep in line with naming through the rest of the repo

Reply:
Yes!! I did it